### PR TITLE
fix: Update aws-lambda-instrumentation to SDK v0.25.0

### PR DESCRIPTION
**Pull request by willarmiros**
## Which problem is this PR solving?

- https://github.com/open-telemetry/opentelemetry-js-contrib/pull/635#issuecomment-904882232

## Short description of the changes

- Upgrading the opentelemetry-instrumentation-aws-lambda package's core dependencies to `0.25.0`, and fixing associated breaking changes.

### CODE REVIEW

**Review by Flarna — APPROVED** [review]

**Comment by willarmiros** [comment]
Converting to draft since this doesn't fix the bug (yet). Just wanted to test it with GH Actions, but am reproducing locally now.

**Comment by willarmiros** [comment]
Now that #661 is merged this is ready for review :+1:

File: plugins/node/opentelemetry-instrumentation-aws-lambda/test/integrations/lambda-handler.force-flush.test.ts [hunk 1]
```diff
@@ -23,8 +23,8 @@ import { AwsLambdaInstrumentation } from '../../src';
 import {
   BatchSpanProcessor,
   InMemorySpanExporter,
-} from '@opentelemetry/tracing';
-import { NodeTracerProvider } from '@opentelemetry/node';
+} from '@opentelemetry/sdk-trace-base';
```
Comment (dyladan) [inline plugins/node/opentelemetry-instrumentation-aws-lambda/test/integrations/lambda-handler.force-flush.test.ts:26]: This dependency is missing in package.json dependencies

**Review by dyladan — CHANGES_REQUESTED** [review]

File: plugins/node/opentelemetry-instrumentation-aws-lambda/test/integrations/lambda-handler.force-flush.test.ts (hunk 1 above)
Comment (longility) [reply plugins/node/opentelemetry-instrumentation-aws-lambda/test/integrations/lambda-handler.force-flush.test.ts:26]: This is in `devDependencies`. Basically, if the tests passes, then we are good as the dependencies are good.

Comment (dyladan) [reply plugins/node/opentelemetry-instrumentation-aws-lambda/test/integrations/lambda-handler.force-flush.test.ts:26]: It was not, but it has been added. It was a transitive dependency of something else which is why the tests were passing but it is good practice to declare all of your dependencies.

**Review by dyladan — APPROVED** [review]

File: plugins/node/opentelemetry-instrumentation-aws-lambda/test/integrations/lambda-handler.force-flush.test.ts (hunk 1 above)
Comment (longility) [reply plugins/node/opentelemetry-instrumentation-aws-lambda/test/integrations/lambda-handler.force-flush.test.ts:26]: Apparently, I'm blind. Thanks for sharing your perspective.

**Pull request merged**